packages/gatsby/src/utils/adapter/manager.ts

```diff
@@ -334,6 +337,10 @@ function getRoutesManifest(): {
       route.path = `/${route.path}`
     }
 
+    if (pathPrefix && !route.path.startsWith(pathPrefix)) {
```

Comment:
@pieh - I believe this will break external paths (e.g. https://www.foo.com) which were recently fixed in #38646.